### DISCUSSION TITLE
Please troubleshoot my workflow for highlighting and commenting directly to PDF and copying that highlighted text and comment to note

### DISCUSSION THREAD

**Discussion by 7thgenerationdesign** [RyotaUshio/obsidian-pdf-plus#264] [Q&A]
First off, my hat is off to the developer for an absolutely amazing plugin.

I want to best streamline the workflow for:

- adding a highlighting a section of text directly in PDF
- adding a comment to that highlighted text directly in PDF
- adding a link to the highlighted text, quote with highlighted text, and the comment to a callout in a note that I have open

My current workflow (I've already updated my template for "Quote in callout" with {{comment}} to include the annotation comment in my link to selection):

1. create new note and have open in background
2. open PDF that has been imported into Obsidian that I want to annotate and takes notes on
3. enable "Add highlights to file directly"
4. select text in PDF that I want to add highlight and comment to
5. left click on highlight color in toolbar to add highlight to selected text
6. left click on highlighted text, and then click "edit" to bring up "Edit Annotation Contents" dialog
7. add comment to "Contents". click save.
8. left click on highlighted text again, and then click "copy" icon

At that point, the PDF file has been directly annotated, and the highlighted text and associated comment has been copied into my note.

This certainly gets me to my end goal, I'm wondering if I am missing a way that this could be streamlined however.

I did just discover the option of adding a hotkey to "Extract and Copy Annotations in this PDF", so I see that I could eliminate Step 8 if I was okay with just extracting and copying all annotations that I've made at the end, however I imagine there are moments when I'll want to copy each annotations (with associated comment) to a note as I go, rather than all at once at the end, and I'd love to know if there's a way I can streamline my process further.

Thanks in advance!

**Comment by RyotaUshio** [comment]
Hi, thanks for the kind words!

Unfortunately, I don't have an idea how we can improve your workflow upon the current level.
This plugin does supports adding & editing highlights directly to PDFs, but I must admit it's quite limited.
At this point, your workflow is probably the best you can get.

(You might want to check out this remark, although this does not really answer your question:
https://github.com/RyotaUshio/obsidian-pdf-plus?tab=readme-ov-file#note-on-saving-annotations-directly-in-pdf)

**Comment by ryanjamurphy** [comment]
I'm also interested in this workflow! Ideally, for me, it would work like this (when configured to work this way, of course).

Enable "add highlights to file directly" for all PDFs. Then, in any given PDF:

1. Select text to highlight
2. Tap the highlight color to add a highlight.
3. The annotation box automatically shows with the cursor focused on the comment box
4. Type out your comment
5. Hit "save"
6. Select "copy" on that annotation to copy it in-template to your annotations file

Arguably step 5 could also be "save and copy," allowing you to skip step 6.

Thanks for PDF++! Such a fantastic tool.
